padinfo/boardgenerator.py

```diff
@@ -0,0 +1,47 @@
+class BoardGenerator(object):
+    """Generate a Dawnglare link with optional inversion."""
+
+    allowed_letters = "RGBLDHXJMPZ"
+    inversion = "-1"
+    dawnglare_link = "https://pad.dawnglare.com/?patt={}&showfill=1"
+    board_sizes = {
+        20: {"height": 4, "width": 5},
+        30: {"height": 5, "width": 6},
+        42: {"height": 6, "width": 7}
+    }
+
+    def __init__(self, board):
+        self.board = board.replace(" ", "")
+        self.invert = self.inversion in self.board
+        self.board = self.board.replace(self.inversion, "")
+        self.board_length = len(self.board)
+        self.invalid_board = self.board_length not in self.board_sizes
+        self.invalid_orbs = not all(letter in self.allowed_letters for letter in self.board)
+        if not self.invalid_board and not self.invalid_orbs:
```

Comment:
you can do a happy-path return here too - if there ARE errors then return, otherwise de-indent and continue doing stuff